src/helm/benchmark/compute_contamination_metrics.py

```diff
@@ -0,0 +1,212 @@
+    """
+    A memory-efficient class for contamination metrics. The core data structures are bit arrays where
+    every bit records whether an instance is dirty (contaminated) or not.
+    """
+
+    def __init__(self, scenario_name: str, num_instances: int):
+        self.scenario_name = scenario_name
+        self.num_instances = num_instances
+        self._input_bits = bitarray(num_instances)
+        self._reference_bits = bitarray(num_instances)
+        self._input_bits.setall(0)
+        self._reference_bits.setall(0)
+
+    @classmethod
+    def from_scenario(cls, scenario: LightScenario):
+        return cls(scenario_name=scenario.name, num_instances=len(scenario.light_instances))
+
+    def write_dirty(self, instance_id: int, part: str):
+        if part == "input":
+            self._input_bits[instance_id] = 1
+        elif part == "ref":
+            self._reference_bits[instance_id] = 1
+        else:
+            raise ValueError(f"There is no valid part of instance named {part}")
+
+    def get_bit(self, instance_id: int, part: str) -> int:
+        if part == "input":
+            return self._input_bits[instance_id]
+        elif part == "ref":
+            return self._reference_bits[instance_id]
+        else:
+            raise ValueError(f"There is no valid part of instance named {part}")
+
+    def merge(self, metric):
+        """Merge two metric instance of the same scenario"""
+        if self.scenario_name != metric.scenario_name:
+            raise ValueError("Only metrics for the same scenario can be merged.")
+        if self.num_instances != metric.num_instances:
+            raise ValueError("The sizes of the two scenarios need to equal.")
+        self._input_bits |= metric._input_bits
+        self._reference_bits |= metric._reference_bits
+
+    @property
+    def num_input_positive_instances(self):
+        return self._input_bits.count()
+
+    @property
+    def num_reference_positive_instances(self):
+        return self._reference_bits.count()
+
+    @property
+    def input_positive_rate(self):
+        return self._input_bits.count() / self.num_instances
+
+    @property
+    def reference_positive_rate(self):
+        return self._reference_bits.count() / self.num_instances
+
+    def generate_summary(self, tags) -> str:
+        """Output a summary of the metric"""
+        summary = {
+            "Setting": f"{self.scenario_name},{','.join(tags)}",
+            "total_instances": self.num_instances,
+            "num_input_positive_instances": self.num_input_positive_instances,
+            "num_reference_positive_instances": self.num_reference_positive_instances,
+            "input_positive_rate": self.input_positive_rate,
+            "reference_positive_rate": self.reference_positive_rate,
+        }
+        return json.dumps(summary)
+
+
+def load_scenarios_from_jsonl(filename: str) -> List[LightScenario]:
+    """
+    Create a list of light scenarios from a jsonl file.
```

Comment:
Make this definition more precise (jsonl file doesn't specify what the dataclass is).